src/Browser.php

```diff
@@ -784,7 +880,55 @@ private function requestMayBeStreaming($method, $url, array $headers = array(),
         }
 
         return $this->transaction->send(
-            new Request($method, $url, $headers, $body, $this->protocolVersion)
+            new Request($method, $url, $this->mergeHeaders($headers), $body, $this->protocolVersion)
```

Comment:
I see why this was done, but perhaps we can inline the helper method into this method given it is not reused anywhere and is quite narrow in scope?

Reply:
i inlined it into the method and removed the mergeHeaders($headers) method